assets/scripts/oauth-services-api/index.js

```diff
@@ -0,0 +1,49 @@
+import page from 'page'
+import store from './../store.js'
+import { GitHubAPI } from './github.js'
+
+import './../types.js'
+
+export class OAuthProvider {
```

Comment:
Chuis un peu intrigué par ce fichier
J'ai l'impression que ça pourrait être écrit plus simplement, mais ptèt il me manque du contexte

On peut en repapoter de manière synchrone

Reply:
On en a papoté et en gros, ça pourrait être uniquement une fonction genre :

```js
/**
 * @typedef { (type: 'github', options: GithubOptions) => OAuthServiceAPI} GithubOauthAPIServiceMaker
 * @typedef { (type: 'gitlab', options: GitlabOptions) => OAuthServiceAPI} GitlabOauthAPIServiceMaker
 */

/** 
 * @type { GithubOauthAPIServiceMaker | GitlabOauthAPIServiceMaker } 
 * 
 * @param {string} type
 * @param {any} options
 */
const makeOauthProvider = (type, options) => {
  if(type === 'github')
  // ...
}
```

(j'ai pas réussi à faire + simple pour les définitions)

On va aussi laisser la gestion du localStorage dans le store et mettre un commentaire pour expliquer pourquoi (et donc ce module n'exporte qu'une fonction et est state-less)

Reply:
Ah, je me suis raté
Il existe une syntaxe dédiée pour les overload de fonction : https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#overload-support-in-jsdoc
Ça vaut le coup de plutôt utiliser ça